python/ray/autoscaler/_private/gcp/node_provider.py

```diff
@@ -164,14 +164,23 @@ def internal_ip(self, node_id: str):
             return ip
 
     @_retry
-    def create_node(self, base_config: dict, tags: dict, count: int) -> None:
+    def create_node(self, base_config: dict, tags: dict, count: int) -> Dict[str, dict]:
+        """Creates instances.
+
+        Returns dict mapping instance id to each create operation result for the created
+        instances.
+        """
         with self.lock:
             labels = tags  # gcp uses "labels" instead of aws "tags"
 
             node_type = get_node_type(base_config)
             resource = self.resources[node_type]
 
-            resource.create_instances(base_config, labels, count)
+            results = resource.create_instances(base_config, labels, count)  # type: List[Tuple[dict, str]] 
+            instance_dict: Dict[str, dict] = {}
+            for result, instance_id in results:
+                instance_dict[instance_id] = result
+            return instance_dict
```

Comment:
Hey @hanming-lu, did you see my suggestion? We're trying to improve craftsmanship of our code, and while what you have is good, further simplification is even better. What do you think of removing this for loop and variable creation to this single dict comprehension?
```python
return {instance_id: result for result, instance_id in results}
```

Reply:
My apologies, I missed the comment. I have just updated it :)